balancer/pickfirst/pickfirst.go

```diff
@@ -35,19 +35,26 @@ import (
 	"google.golang.org/grpc/serviceconfig"
 )
 
-func init() {
-	balancer.Register(pickfirstBuilder{})
-	internal.ShuffleAddressListForTesting = func(n int, swap func(i, j int)) { rand.Shuffle(n, swap) }
-}
-
-var logger = grpclog.Component("pick-first-lb")
+var (
+	logger      = grpclog.Component("pick-first-lb")
+	shuffleFunc = rand.Shuffle
+)
 
 const (
 	// Name is the name of the pick_first balancer.
 	Name      = "pick_first"
 	logPrefix = "[pick-first-lb %p] "
 )
 
+func init() {
+	balancer.Register(pickfirstBuilder{})
+	internal.ShuffleAddressListForTesting = func(sf func(n int, swap func(i, j int))) func(n int, swap func(i, j int)) {
```

Comment:
Renamed to `SetRandShuffleForTesting` after moving the variable into `pickfirst/internal` omitting `PickFirst` because it's now in the `pickfirst` directory.